Authentication/Login/ViewModel/LoginViewModel.swift

```diff
@@ -88,12 +92,25 @@ public final class LoginViewModel<User, SessionService: SessionServiceType> : Lo
         
         emailValidationErrors = Property(initial: [], then: emailValidationResult.map { $0.errors })
         passwordValidationErrors = Property(initial: [], then: passwordValidationResult.map { $0.errors })
+        logInProviderUserSignal = Signal.merge(providerUserSignals)
     }
     
 }
 
 fileprivate extension LoginViewModel {
     
+    fileprivate func initializeLogInSuccesfulSignal() -> Signal<(), NoError> {
+        let loggedInWithProviderUserSignal = self.logInProviderUserSignal.flatMap(.latest) { [unowned self] loginProviderUserType -> SignalProducer<User, SessionServiceError> in
+            return self._sessionService.logIn(withUserType: loginProviderUserType)
```

Comment:
I think I prefer `login(withUser: loginProviderUser)` since it's not just a type.